File: frontend/src/types/loyaltyPoint.ts
export enum LoyaltyPointType {
  EARNED = "EARNED",
  REDEEMED = "REDEEMED",
  RESTORED = "RESTORED",
  EXPIRED = "EXPIRED",
}

export interface LoyaltyPoint {
  id: string;
  customerId: string;
  points: number;
  description: string;
  type: LoyaltyPointType;
  amount: number;
  createdDate: string;
}

File: commerce-service/src/main/java/com/ecommerce/commerce_service/model/LoyaltyPoint.java
package com.ecommerce.commerce_service.model;

import com.ecommerce.common.model.AdvanceBaseModal;
import lombok.*;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.UUID;

@Entity(name = "loyalty_points")
@Table
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class LoyaltyPoint extends AdvanceBaseModal {

    private UUID customerId;

    private Integer points;

    private String description;

    @Enumerated(EnumType.STRING)
    private LoyaltyPointType type;

    @Column(precision = 19, scale = 2)
    private BigDecimal amount;
}
